### PR TITLE
[SystemZ] Add extended mnemonics tests.

### PR DESCRIPTION
**Pull request by dominik-steenken**
This adds tests for some extended mnemonics of load, branch, and compare-and-trap instructions.

### CODE REVIEW

**Comment by dominik-steenken** [comment]
@uweigand FYI

File: llvm/test/MC/SystemZ/insn-good-z13.s [hunk 1]
```diff
@@ -190,6 +203,19 @@
         locghino %r11, 32512
         locghi %r11, 32512, 15
 
+#CHECK: locghim %r10, 2766 # encoding: [0xec,0xa4,0x0a,0xce,0x00,0x46]
+#CHECK: locghinm %r11, 6862 # encoding: [0xec,0xbb,0x1a,0xce,0x00,0x46]
+#CHECK: locghip %r12, 10958 # encoding: [0xec,0xc2,0x2a,0xce,0x00,0x46]
+#CHECK: locghiz %r13, 15054 # encoding: [0xec,0xd8,0x3a,0xce,0x00,0x46]
+#CHECK: locghinp %r14, 19150 # encoding: [0xec,0xed,0x4a,0xce,0x00,0x46]
+#CHECK: locghinz %r15, 23246 # encoding: [0xec,0xf7,0x5a,0xce,0x00,0x46]
```
Comment (uweigand) [inline llvm/test/MC/SystemZ/insn-good-z13.s:211]: Please try to following the spacing/indentation with the other lines (the "encoding:" should ideally line up).  Here and everywhere ...

File: llvm/test/MC/SystemZ/insn-good-z14.s [hunk 2]
```diff
@@ -87,6 +88,7 @@
 	bic	0, 524287(%r1,%r15)
 	bic	0, 524287(%r15,%r1)
 	bic	15, 0
+	bi	0
```
Comment (uweigand) [inline llvm/test/MC/SystemZ/insn-good-z14.s:91]: We already have tests for plain `bi`, so this seems redundant.

File: llvm/test/MC/SystemZ/insn-good-z14.s [hunk 3]
```diff
@@ -108,33 +112,39 @@
...
 #CHECK:	bil	0(%r15)			# encoding: [0xe3,0x40,0xf0,0x00,0x00,0x47]
+#CHECK:	bim	0(%r13)			# encoding: [0xe3,0x40,0xd0,0x00,0x00,0x47]
 
 	bic	4, 0(%r7)
 	bil	0(%r15)
+	bim	0(%r13)
 
 #CHECK:	bic	5, 0(%r7)		# encoding: [0xe3,0x50,0x70,0x00,0x00,0x47]
-#CHECK:	binhe	0(%r15)			# encoding: [0xe3,0x50,0xf0,0x00,0x00,0x47]
+#CHECK:	binhe	0(%r15)		# encoding: [0xe3,0x50,0xf0,0x00,0x00,0x47]
```
Comment (uweigand) [inline llvm/test/MC/SystemZ/insn-good-z14.s:122]: See above w.r.t. spaces.  Here this may be a tab vs. spaces issue?

**Review by uweigand — COMMENTED** [review]
Looks good to me, with a few minor comments inline.  I assume you verified that the GNU assembler agrees with these encodings as well?

File: llvm/test/MC/SystemZ/insn-good-z13.s (hunk 1 above)
Comment (dominik-steenken) [reply llvm/test/MC/SystemZ/insn-good-z13.s:211]: I will add the missing whitespace.

File: llvm/test/MC/SystemZ/insn-good-z14.s (hunk 2 above)
Comment (dominik-steenken) [reply llvm/test/MC/SystemZ/insn-good-z14.s:91]: I agree, will delete.

File: llvm/test/MC/SystemZ/insn-good-z14.s (hunk 3 above)
Comment (dominik-steenken) [reply llvm/test/MC/SystemZ/insn-good-z14.s:122]: NKind of. In this case, my editor assumed a tab was 4 spaces long, while this assumes a tab is 8 spaces long. I've reconfigured my editor for 8 spaces, and have realigned the tabs in this section.

**Review by uweigand — APPROVED** [review]
LGTM now, thanks!

**Comment by dominik-steenken** [comment]
> I assume you verified that the GNU assembler agrees with these encodings as well?
I did, yes.

**Pull request merged**